Core/Object Arts/Dolphin/IDE/Base/SmalltalkSearchPresenter.cls

```diff
@@ -154,47 +151,47 @@ rangeToCompleteAt: anInteger
 				ifFalse: [anInteger to: view caretPosition - 1]]!
 
 searchItemsStartingWith: aString maxItems: anInteger
```

Comment:
Sure:

* As you noticed, `isLowercase ifTrue:` -> `isUppercase ifFalse:`—treat searches starting with punctuation as referring to selectors.
* Add packages to the list ahead of classes, reflecting their higher priority when the search is actually performed.
* Scan `Smalltalk` for source and alias globals and add them to the list also.
* Actually respect the `maxItems:` parameter for non-selector searches.
* Revert to using string equality to compare class names, as _metaclass_ names are strings, not symbols.